android/app/src/main/java/org/fossasia/openevent/core/notifications/NotificationsFragmentViewModel.java

```diff
@@ -0,0 +1,65 @@
+package org.fossasia.openevent.core.notifications;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.common.arch.LiveRealmData;
+import org.fossasia.openevent.core.notifications.repository.NotificationsRepository;
+import org.fossasia.openevent.data.Notification;
+import org.fossasia.openevent.data.repository.RealmDataRepository;
+
+import java.util.List;
+
+import io.reactivex.Observable;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import timber.log.Timber;
+
+public class NotificationsFragmentViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+    private LiveData<List<Notification>> notificationsData;
+    private MutableLiveData<Boolean> notificationDownloadResponse;
+    private final CompositeDisposable compositeDisposable;
+    private final NotificationsRepository notificationsRepository;
+
+    public NotificationsFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+        notificationsRepository = new NotificationsRepository();
+    }
+
+    public LiveData<List<Notification>> getNotificationsData() {
+        if (notificationsData == null) {
+            LiveRealmData<Notification> liveRealmData = RealmDataRepository.asLiveData(realmRepo.getNotifications());
+            notificationsData = Transformations.map(liveRealmData, input -> input);
+        }
+        return notificationsData;
+    }
+
+    public LiveData<Boolean> downloadNotifications() {
+        if (notificationDownloadResponse == null) {
+            notificationDownloadResponse = new MutableLiveData<>();
+        }
+        Observable<List<Notification>> notificationsObservable = notificationsRepository.downloadNotifications();
+        if (notificationsObservable != null) {
+            compositeDisposable.add(notificationsObservable.subscribe(notifications -> {
+                Timber.i("Downloaded notifications");
+                notificationsRepository.saveNotifications(notifications);
```

Comment:
This should be responsibility of NotificationsRespository

Reply:
Also add screenshot of what happens if there are no notifications

Reply:
> This should be responsibility of NotificationsRespository

Sorry, i couldn't get you. It is calling the repository only.

Reply:
Did you mean chaining the call ?

Reply:
No, just use `onNext` in the repository to save the notifications in db